statsmodels/base/tests/test_penalized.py

```diff
@@ -0,0 +1,728 @@
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+
+        modp = Logit(y, x)
+        cls.res2 = modp.fit(disp=0)
+
+        mod = LogitPenalized(y, x, penal=cls.penalty)
+        mod.pen_weight = 0
+        cls.res1 = mod.fit(disp=0)
+
+        cls.atol = 1e-4  # why not closer ?
+
+
+class TestPenalizedLogitOracle(CheckPenalizedLogit):
+    # TODO: check, adjust cov_type
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        modp = Logit(y, x[:, :cls.k_nonzero])
+        cls.res2 = modp.fit(disp=0)
+
+        mod = LogitPenalized(y, x, penal=cls.penalty)
+        mod.pen_weight *= .5
+        mod.penal.tau = 0.05
+        cls.res1 = mod.fit(method='bfgs', maxiter=100, disp=0)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+
+        cls.atol = 5e-3
+
+
+class TestPenalizedGLMLogitOracle(CheckPenalizedLogit):
+    # TODO: check, adjust cov_type
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        modp = GLM(y, x[:, :cls.k_nonzero], family=family.Binomial())
+        cls.res2 = modp.fit(disp=0)
+
+        mod = GLMPenalized(y, x, family=family.Binomial(), penal=cls.penalty)
+        mod.pen_weight *= .5
+        mod.penal.tau = 0.05
+        cls.res1 = mod.fit(method='bfgs', maxiter=100, disp=0)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+
+        cls.atol = 5e-3
+
+
+class TestPenalizedLogitOraclePenalized(CheckPenalizedLogit):
+    # TODO: check, adjust cov_type
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        modp = LogitPenalized(y, x[:, :cls.k_nonzero], penal=cls.penalty)
+        cls.res2 = modp.fit(method='bfgs', maxiter=100, disp=0)
+
+        mod = LogitPenalized(y, x, penal=cls.penalty)
+        #mod.pen_weight *= 1.5
+        #mod.penal.tau = 0.05
+        cls.res1 = mod.fit(method='bfgs', maxiter=100, trim=False, disp=0)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+
+        cls.atol = 1e-3
+
+
+class TestPenalizedLogitOraclePenalized2(CheckPenalizedLogit):
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        modp = LogitPenalized(y, x[:, :cls.k_nonzero], penal=cls.penalty)
+        modp.pen_weight *= 0.5
+        modp.penal.tau = 0.05
+        cls.res2 = modp.fit(method='bfgs', maxiter=100, disp=0)
+
+        mod = LogitPenalized(y, x, penal=cls.penalty)
+        mod.pen_weight *= 0.5
+        mod.penal.tau = 0.05
+        cls.res1 = mod.fit(method='bfgs', maxiter=100, trim=True, disp=0)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+
+        cls.atol = 1e-8
+        cls.k_params = cls.k_nonzero
+
+    def test_zeros(self):
+
+        # test for trimmed result
+        assert_equal(self.res1.params[self.k_nonzero:], 0)
+        # we also set bse to zero
+        assert_equal(self.res1.bse[self.k_nonzero:], 0)
+
+
+# the following classes are copies of Poisson with model adjustments
```

Comment:
this same comment occurs on L384.  is it accurate/needed in both places?

Reply:
reminder that we might be able to parameterize this better with less code duplication
(but I don't change the structure in this PR.)